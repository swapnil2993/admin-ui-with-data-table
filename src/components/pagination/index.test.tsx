import { fireEvent, render, screen } from "@testing-library/react";
import Pagination from "./index";

describe("Pagination", () => {
  it("renders correctly", () => {
    const paginate = jest.fn();
    const { container } = render(
      <Pagination totalPages={5} paginate={paginate} currentPage={1} />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls paginate function with correct page number when button is clicked", () => {
    const paginateMock = jest.fn();
    render(
      <Pagination totalPages={5} paginate={paginateMock} currentPage={1} />
    );

    fireEvent.click(screen.getByText("2"));
    expect(paginateMock).toHaveBeenCalledWith(2);

    fireEvent.click(screen.getByText(">>"));
    expect(paginateMock).toHaveBeenCalledWith(5);
  });

  it("disables buttons when for first page", () => {
    const paginateMock = jest.fn();
    render(
      <Pagination totalPages={5} paginate={paginateMock} currentPage={1} />
    );

    expect(screen.getByText("<<")).toBeDisabled();
    expect(screen.getByText("<")).toBeDisabled();
    expect(screen.getByText(">")).not.toBeDisabled();
    expect(screen.getByText(">>")).not.toBeDisabled();
  });

  it("disables buttons when appropriate for last page", () => {
    const paginateMock = jest.fn();
    render(
      <Pagination totalPages={3} paginate={paginateMock} currentPage={3} />
    );

    expect(screen.getByText("<<")).not.toBeDisabled();
    expect(screen.getByText("<")).not.toBeDisabled();
    expect(screen.getByText(">")).toBeDisabled();
    expect(screen.getByText(">>")).toBeDisabled();
  });

  it("highlights active page", () => {
    const paginateMock = jest.fn();
    render(
      <Pagination totalPages={5} paginate={paginateMock} currentPage={3} />
    );

    expect(screen.getByText("3")).toHaveClass("active");
  });
});
