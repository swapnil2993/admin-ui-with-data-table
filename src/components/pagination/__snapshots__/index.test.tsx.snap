// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div>
  <nav
    class="pagination-wrapper"
  >
    <div
      class="pagination"
    >
      <button
        class="page-btn-skip first-page"
        disabled=""
      >
        &lt;&lt;
      </button>
      <button
        class="page-btn-skip previous-page"
        disabled=""
      >
        &lt;
      </button>
      <button
        class="page-btn active"
      >
        1
      </button>
      <button
        class="page-btn"
      >
        2
      </button>
      <button
        class="page-btn"
      >
        3
      </button>
      <button
        class="page-btn"
      >
        4
      </button>
      <button
        class="page-btn"
      >
        5
      </button>
      <button
        class="page-btn-skip next-page"
      >
        &gt;
      </button>
      <button
        class="page-btn-skip last-page"
      >
        &gt;&gt;
      </button>
    </div>
  </nav>
</div>
`;
